sky/spot/recovery_strategy.py

```diff
@@ -116,6 +116,11 @@ def terminate_cluster(self, max_retry: int = 3) -> None:
                                        f'{self.cluster_name}.') from e
                 logger.error('Failed to terminate the spot cluster '
                              f'{self.cluster_name}. Retrying.')
+                import traceback  # pylint: disable=import-outside-toplevel
+                logger.error(f'  Detailed exception: {e}')
+                logger.info(f'  Traceback: {traceback.format_exc()}')
```

Comment:
IIRC, the existing usage of `format_exc` in this module didn't print out the stacktrace. Could we double check?

Reply:
I think we do need `format_exc` to print out the full trace. I tried one simple example below
```
def func():
    raise ValueError("test")
    
try:
    func()
except Exception as e:
    print(f'{type(e)}: {str(e)}')
    print('===full trace===')
    import traceback
    print(traceback.format_exc())
```
and got output
```
<class 'ValueError'>: test
===full trace===
Traceback (most recent call last):
  File "sky/test_err.py", line 5, in <module>
    func()
  File "sky/test_err.py", line 2, in func
    raise ValueError("test")
ValueError: test
```

Reply:
Could we try a real `spot launch` example? IIRC, that didn't produce a usable trace.

Reply:
I comment out this line and the entire trace will then show
https://github.com/skypilot-org/skypilot/blob/07ff922f8a2b73c6ddc533009527944840e27757/sky/utils/ux_utils.py#L33
```
import sky

try:
    task = sky.Task(name='x')
    t = sky.Task(run='ls')
    t.set_resources(sky.Resources(sky.GCP()))
    sky.launch(task=t, cluster_name='xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx')
except Exception as e:
    print(f'{type(e)}: {str(e)}')
    print('===full trace===')
    import traceback
    print(traceback.format_exc())
```
Output:
```
...
<class 'sky.exceptions.InvalidClusterNameError'>: Cluster name 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' has 60 chars; maximum length is 35 chars on GCP.
===full trace===
Traceback (most recent call last):
  File "test_err.py", line 11, in <module>
    sky.launch(task=t, cluster_name='xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx')
  File "/home/gcpuser/sky_workdir/skypilot/sky/utils/common_utils.py", line 241, in _record
    return f(*args, **kwargs)
  File "/home/gcpuser/sky_workdir/skypilot/sky/utils/common_utils.py", line 241, in _record
    return f(*args, **kwargs)
  File "/home/gcpuser/sky_workdir/skypilot/sky/execution.py", line 411, in launch
    _execute(
  File "/home/gcpuser/sky_workdir/skypilot/sky/execution.py", line 260, in _execute
    handle = backend.provision(task,
  File "/home/gcpuser/sky_workdir/skypilot/sky/utils/common_utils.py", line 241, in _record
    return f(*args, **kwargs)
  File "/home/gcpuser/sky_workdir/skypilot/sky/utils/common_utils.py", line 220, in _record
    return f(*args, **kwargs)
  File "/home/gcpuser/sky_workdir/skypilot/sky/backends/backend.py", line 49, in provision
    return self._provision(task, to_provision, dryrun, stream_logs,
  File "/home/gcpuser/sky_workdir/skypilot/sky/backends/cloud_vm_ray_backend.py", line 2129, in _provision
    to_provision_config = self._check_existing_cluster(
  File "/home/gcpuser/sky_workdir/skypilot/sky/utils/common_utils.py", line 241, in _record
    return f(*args, **kwargs)
  File "/home/gcpuser/sky_workdir/skypilot/sky/backends/cloud_vm_ray_backend.py", line 3231, in _check_existing_cluster
    task_cloud.check_cluster_name_is_valid(cluster_name)
  File "/home/gcpuser/sky_workdir/skypilot/sky/clouds/cloud.py", line 408, in check_cluster_name_is_valid
    raise exceptions.InvalidClusterNameError(
sky.exceptions.InvalidClusterNameError: Cluster name 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' has 60 chars; maximum length is 35 chars on GCP.
```
Was it because of `sys.tracebacklimit = 0` so it didn't produce trace?